### DISCUSSION TITLE
Questions about the pybricks environment

### DISCUSSION THREAD

**Discussion by techcatgato** [pybricks/support#1707] [Q&A]
Since pybricks isn't just regular python, I have 2 questions:
1. Can I use any library I want aside the pybrick ones?
2. When I code using pybricks and use file-managing functions such as DataLog, where are the files stored?

**Comment by laurensvalk** [comment]
> Can I use any library I want aside the pybrick ones?

Is this for EV3? Since Pybricks uses MicroPython, you cannot install regular Python libraries. However, if you have the source files of a library and the library does not depend on other libraries, you can include it with your project files.

> When I code using pybricks and use file-managing functions such as DataLog, where are the files stored?

On EV3, files are stored in the same folder as the script. You can upload them back to your computer as described here: https://pybricks.com/ev3-micropython/startrun.html#managing-files-on-the-ev3-brick

**Reply by techcatgato** [reply]
Talking about spike prime/robot inventor

**Reply by laurensvalk** [reply]
You can include other files and import them. They will be uploaded to the hub along with your main program.

These hubs don't have a file system for saving files or logging data.

**Reply by techcatgato** [reply]
> You can include other files and import them. They will be uploaded to the hub along with your main program.
> 
> These hubs don't have a file system for saving files or logging data.

Heres my idea, tell me if its possible - The data log will be stored on a pc where ill use a regular python code to translate it into the data i need in the main robot program
